processor/src/main/java/org/mapstruct/ap/internal/model/common/Parameter.java

```diff
@@ -95,25 +95,28 @@ public boolean isMappingContext() {
     }
 
     @Override
-    public int hashCode() {
-        int hash = 5;
-        hash = 23 * hash + (this.name != null ? this.name.hashCode() : 0);
-        return hash;
-    }
-
-    @Override
-    public boolean equals(Object obj) {
-        if ( obj == null ) {
-            return false;
+    public boolean equals(Object o) {
+        if ( this == o ) {
+            return true;
         }
-        if ( getClass() != obj.getClass() ) {
+        if ( o == null || getClass() != o.getClass() ) {
             return false;
         }
-        final Parameter other = (Parameter) obj;
-        if ( (this.name == null) ? (other.name != null) : !this.name.equals( other.name ) ) {
+
+        Parameter parameter = (Parameter) o;
+
+        if ( name != null ? !name.equals( parameter.name ) : parameter.name != null ) {
             return false;
         }
-        return true;
+        return type != null ? type.equals( parameter.type ) : parameter.type == null;
+
+    }
+
+    @Override
+    public int hashCode() {
```

Comment:
I usually delete equals/hashCode and generate them from the IntelliJ template instead of adding needed lines, I did not notice the methods were swapped.